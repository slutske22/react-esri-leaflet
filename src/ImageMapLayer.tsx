import * as React from "react";
import Layer from "./EsriLeafletLayer";
import * as EsriLeaflet from "esri-leaflet";
import { RasterLayerOptions } from "esri-leaflet";

interface Props extends RasterLayerOptions {
  url: string;
  format?: string;
  bandIds?: string;
  noData?: number;
  noDataInterpretation?: string;
  pixelType?: string;
  renderingRule?: object;
  mosaicRule?: object;
  token?: string;
  proxy?: string;
  useCors?: boolean;
  to?: Date;
  from?: Date;
}

/**
 * ImageMapLayer component, react-leaflet wrapper for [L.esri.ImageMapLayer](https://developers.arcgis.com/esri-leaflet/api-reference/layers/image-map-layer/).
 *
 * For more info: https://github.com/slutske22/react-esri-leaflet#components
 */
const ImageMapLayer = React.forwardRef<EsriLeaflet.RasterLayer, Props>(
  (props: Props, ref) => (
    <Layer ref={ref} layerType="imageMapLayer" {...props} />
  )
);

export default ImageMapLayer;
