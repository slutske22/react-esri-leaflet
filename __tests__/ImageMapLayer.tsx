import React from "react";
import { MapContainer } from "react-leaflet";
import { RasterLayer } from "esri-leaflet";
import { render } from "@testing-library/react";

import ImageMapLayer from "../src/ImageMapLayer";

describe("ImageMapLayer", () => {
  let mapRef;

  it("creates an instance of esri-leaflet imageMapLayer and adds it to the map", () => {
    render(
      <MapContainer
        ref={(ref) => {
          mapRef = ref;
        }}
        center={[32, -117]}
        zoom={6}
      >
        <ImageMapLayer url="https://landsat.arcgis.com/arcgis/rest/services/Landsat/PS/ImageServer" />
      </MapContainer>
    );

    const addedLayer = Object.values(mapRef._layers)[0];
    expect(addedLayer).toBeInstanceOf(RasterLayer);
  });

  it("creates the same layer every time", () => {
    const { container } = render(
      <MapContainer center={[0, 0]} zoom={10}>
        <ImageMapLayer url="https://landsat.arcgis.com/arcgis/rest/services/Landsat/PS/ImageServer" />
      </MapContainer>
    );

    expect(container).toMatchSnapshot();
  });
  //
});
