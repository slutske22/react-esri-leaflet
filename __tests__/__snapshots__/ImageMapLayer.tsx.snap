// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageMapLayer creates the same layer every time 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      >
        <img
          alt=""
          class="leaflet-image-layer"
          crossorigin=""
          src="https://landsat.arcgis.com/arcgis/rest/services/Landsat/PS/ImageServer/exportImage?bbox=0%2C0%2C0%2C0&size=0%2C0&format=jpgpng&transparent=true&bboxSR=3857&imageSR=3857&f=image"
          style="z-index: 1; opacity: 0; left: 0px; top: 0px; width: 0px; height: 0px;"
        />
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              âˆ’
            </span>
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control esri-truncated-attribution:hover esri-truncated-attribution"
        >
          <a
            href="http://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | Powered by 
          <a
            href="https://www.esri.com"
          >
            Esri
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
